### PR TITLE
Solved day 2

### PR DESCRIPTION
**Pull request by moonglum**
Learned parser combinators from [this great blog post](https://two-wrongs.com/parser-combinators-parsing-for-haskell-beginners.html). This is pretty awesome!

## Task

Given a text file (`Day2.txt`) that contains a list of password policies and passwords that look like this:

```
1-3 a: abcde
```

Where `1` is the first number, `3` is the second number, `a` is the character and `abcde` is the password. The task is to find the amount of passwords that are valid according to the policy. The policy is different between the two tasks:

* **Task 1:** There are between first number and second number of character in the password (in the example above, that would be at least 1 and at most 3 `a`s in `abcde`.
* **Task 2:** There is either character at the position described by the first number or the second number (but not both), where position is 1-indexed.

### CODE REVIEW

File: Day2.hs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+, t2
+, main
+) where
+import qualified Data.Text as Text
+import qualified Data.Text.IO as Text
+import qualified Data.Text.Read as Text
+import Control.Monad (unless)
+import Data.Char (isDigit, isLetter)
+import Data.List (elemIndices)
+import qualified Text.ParserCombinators.ReadP as ReadP
```
Comment (larsrh) [inline Day2.hs:12]: Bold choice to use the built-in parser combinators, but nothing wrong with that :+1:

**Review by larsrh — COMMENTED** [review]
Nothing to complain about, all good!

File: Day2.hs (hunk 1 above)
Comment (moonglum) [reply Day2.hs:12]: Awesome! Out of curiosity: Which library would have been your choice?

Comment (larsrh) [reply Day2.hs:12]: https://hackage.haskell.org/package/megaparsec or https://hackage.haskell.org/package/trifecta, probably.

**Pull request merged**